service/frontend/nexus_http_handler.go

```diff
@@ -0,0 +1,162 @@
+	r.PathPrefix("/api/v1/services/{service}/").HandlerFunc(h.dispatchNexusTaskByService)
+}
+
+func (h *NexusHTTPHandler) writeNexusFailure(writer http.ResponseWriter, statusCode int, failure *nexus.Failure) {
+	var err error
+	var bytes []byte
+	if failure != nil {
+		bytes, err = json.Marshal(failure)
+		if err != nil {
+			h.logger.Error("failed to marshal failure", tag.Error(err))
+			writer.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		writer.Header().Set("Content-Type", "application/json")
+	}
+
+	writer.WriteHeader(statusCode)
+
+	if _, err := writer.Write(bytes); err != nil {
+		h.logger.Error("failed to write response body", tag.Error(err))
+	}
+}
+
+// Handler for /api/v1/namespaces/{namespace}/task-queues/{task_queue}/dispatch-nexus-task
+func (h *NexusHTTPHandler) dispatchNexusTaskByNamespaceAndTaskQueue(w http.ResponseWriter, r *http.Request) {
+	if !h.enabled() {
+		h.writeNexusFailure(w, http.StatusNotFound, &nexus.Failure{Message: "nexus endpoints disabled"})
+		return
+	}
+	// Limit the request body to max allowed Payload size. This is hardcoded to 2MB for headers at the moment.
+	// Content headers are transformed to Payload metadata and contribute to the Payload size as well. A separate
+	// limit is enforced on top of this in the nexusHandler.StartOperation method.
+	r.Body = http.MaxBytesReader(w, r.Body, rpc.MaxNexusAPIRequestBodyBytes)
+
+	var err error
+	vars := mux.Vars(r)
+	nc := nexusContext{
+		// This name does not map to an underlying gRPC service. This format is used since for consistency with
+		// the gRPC API names on which the authorizer - the consumer of this string - may depend.
+		apiName:       "/temporal.api.nexusservice.v1/DispatchNexusTask",
+		namespaceName: vars["namespace"],
+		taskQueue:     vars["task_queue"],
+	}
+
+	var tlsInfo *credentials.TLSInfo
+	if r.TLS != nil {
+		tlsInfo = &credentials.TLSInfo{
+			State:          *r.TLS,
+			CommonAuthInfo: credentials.CommonAuthInfo{SecurityLevel: credentials.PrivacyAndIntegrity},
+		}
+	}
+
+	authInfo := h.auth.GetAuthInfo(tlsInfo, r.Header, func() string {
+		return "" // TODO: support audience getter
+	})
+	if authInfo != nil {
+		nc.claims, err = h.auth.GetClaims(authInfo)
+		if err != nil {
+			h.logger.Error("failed to get claims", tag.Error(err))
+			h.writeNexusFailure(w, http.StatusUnauthorized, &nexus.Failure{Message: "unauthorized"})
+			return
+		}
+		// Make the auth info and claims available on the context.
+		r = r.WithContext(h.auth.EnhanceContext(r.Context(), authInfo, nc.claims))
+	}
+
+	r = r.WithContext(context.WithValue(r.Context(), nexusContextKey{}, nc))
+	// Hard-code to taking the first 8 parts for this route. There doesn't seem to be a better way to retrieve the
+	// prefix from gorilla.
+	parts := strings.Split(r.URL.EscapedPath(), "/")
+	http.StripPrefix("/"+path.Join(parts[:8]...), h.nexusHandler).ServeHTTP(w, r)
```

Comment:
I spent quite some time looking to see if there was a better alternative here, but it doesn't look like there is. Could we at least extract "/api/v1/namespaces/{namespace}/task-queues/{task_queue}/dispatch-nexus-task/" to a variable, count the number of slashes there, and then use that?

Reply:
I found a better way in the next PR.